### PR TITLE
Change description based on audit scores

### PR DESCRIPTION
**Pull request by ev1stensberg**
#2311 

@morrme Since you haven't submited a PR yet, I assume you've forgot about it! 
skjermbilde 2017-06-01 kl 20 53 39
skjermbilde 2017-06-01 kl 20 50 06

### CODE REVIEW

**Comment by ebidel** [comment]
I'm not sure we want to make this change. The audit itself is testing that the `short_name` will not be truncated when displayed on the homescreen. So the the language is correct. 

What's confusing is that when you fail the audit, the description reads the same as if you passed it. That's something we need to do for all audits ( https://github.com/GoogleChrome/lighthouse/issues/1611).

**Comment by ev1stensberg** [comment]
Suspected it, but the issue was too short of context. Do you mind summarizing what this PR should do concretely? Not native English, also better for me to understand with more info! (Besides from the issue that was somewhat unclear to me)

**Comment by ebidel** [comment]
> summarizing what this PR should do concretely

I don't think we should make any changes atm. Instead do https://github.com/GoogleChrome/lighthouse/issues/1611.

Basically, when you fail the audit, the description should read:
X Manifest's `short_name` will be truncated when displayed on the homescreen.

When you pass it, the description should read:
✓ Manifest's `short_name` won't be truncated when displayed on the homescreen.

**Comment by ev1stensberg** [comment]
Up for #1611 , I'll get a PR up by Monday

**Comment by ebidel** [comment]
Oh, I didn't mean you had to tackle https://github.com/GoogleChrome/lighthouse/issues/1611 :). It's a big task.

But if you want to, I'd suggest posting implementation ideas on that bug before submitting a PR. It might be good to discuss how best to approach it...since you'll be touching just about every audit.

**Comment by ev1stensberg** [comment]
So, what is left is the CSS audits and/or other stuff that is  non-binary in the output :) Tell me what you think!

**Comment by ev1stensberg** [comment]
skjermbilde 2017-06-01 kl 23 48 45

**Comment by ev1stensberg** [comment]
Need help with the grammar btw!

**Comment by ev1stensberg** [comment]
@ebidel About to adress byte efficiency & #1611 , how can I test CSS? Doesn't seems to add that to my casual `node ./lighthouse-cli/index.js http://example.com`

File: lighthouse-core/audits/accessibility/color-contrast.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ class ColorContrast extends AxeAudit {
       category: 'Accessibility',
       name: 'color-contrast',
       description: 'Background and foreground colors have a sufficient contrast ratio.',
+      fallbackDescription: 'Background and foreground colors does not have a sufficient contrast ratio.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/color-contrast.js]: s/does/do

File: lighthouse-core/audits/accessibility/definition-list.js [hunk 2]
```diff
@@ -34,6 +34,8 @@ class DefinitionList extends AxeAudit {
       name: 'definition-list',
       description: '`<dl>`\'s contain only properly-ordered `<dt>` and `<dd>` groups, `<script>` ' +
           'or <template> elements.',
+      fallbackDescription: '`<dl>`\'s does not contain only properly-ordered `<dt>` and `<dd>` groups, `<script>` ' +
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/definition-list.js]: s/does/do

File: lighthouse-core/audits/accessibility/frame-title.js [hunk 3]
```diff
@@ -33,6 +33,7 @@ class FrameTitle extends AxeAudit {
       category: 'Accessibility',
       name: 'frame-title',
       description: '`<frame>` or `<iframe>` elements have a title.',
+      fallbackDescription: '`<frame>` or `<iframe>` elements does not have a title.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/frame-title.js]: s/does/do

File: lighthouse-core/audits/accessibility/html-has-lang.js [hunk 4]
```diff
@@ -33,6 +33,7 @@ class HTMLHasLang extends AxeAudit {
       category: 'Accessibility',
       name: 'html-has-lang',
       description: '`<html>` element has a `[lang]` attribute.',
+      fallbackDescription: '`<html>` element has not a `[lang]` attribute.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/html-has-lang.js]: s/has not/does not have

File: lighthouse-core/audits/accessibility/html-lang-valid.js [hunk 5]
```diff
@@ -33,6 +33,7 @@ class HTMLLangValid extends AxeAudit {
       category: 'Accessibility',
       name: 'html-lang-valid',
       description: '`<html>` element has a valid value for its `[lang]` attribute.',
+      fallbackDescription: '`<html>` element has not a valid value for its `[lang]` attribute.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/html-lang-valid.js]: s/has not/does not have

File: lighthouse-core/audits/accessibility/image-alt.js [hunk 6]
```diff
@@ -33,6 +33,7 @@ class ImageAlt extends AxeAudit {
       category: 'Accessibility',
       name: 'image-alt',
       description: 'Image elements have `[alt]` attributes.',
+      fallbackDescription: 'Image elements does not have `[alt]` attributes.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/image-alt.js]: s/does/do

File: lighthouse-core/audits/accessibility/input-image-alt.js [hunk 7]
```diff
@@ -33,6 +33,7 @@ class InputImageAlt extends AxeAudit {
       category: 'Accessibility',
       name: 'input-image-alt',
       description: '`<input type="image">` elements have `[alt]` text.',
+      fallbackDescription: '`<input type="image">` elements does not have `[alt]` text.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/input-image-alt.js]: s/does/do

File: lighthouse-core/audits/accessibility/label.js [hunk 8]
```diff
@@ -33,6 +33,7 @@ class Label extends AxeAudit {
       category: 'Accessibility',
       name: 'label',
       description: 'Form elements have associated labels.',
+      fallbackDescription: 'Form elements does not have associated labels.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/label.js]: s/does/do

File: lighthouse-core/audits/accessibility/layout-table.js [hunk 9]
```diff
@@ -35,6 +35,8 @@ class LayoutTable extends AxeAudit {
       name: 'layout-table',
       description: 'Presentational `<table>` elements avoid using `<th>`, `<caption>` or the ' +
           '`[summary]` attribute.',
+      fallbackDescription: 'Presentational `<table>` elements does not avoid using `<th>`, `<caption>` or the ' +
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/layout-table.js]: s/does/do

File: lighthouse-core/audits/accessibility/link-name.js [hunk 10]
```diff
@@ -33,6 +33,7 @@ class LinkName extends AxeAudit {
       category: 'Accessibility',
       name: 'link-name',
       description: 'Links have a discernable name.',
+      fallbackDescription: 'Links does not have a discernable name.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/link-name.js]: s/does/do

File: lighthouse-core/audits/accessibility/list.js [hunk 11]
```diff
@@ -34,6 +34,8 @@ class List extends AxeAudit {
       name: 'list',
       description: 'Lists contain only `<li>` elements and script supporting elements ' +
           '(`<script>` and `<template>`).',
+      fallbackDescription: 'Lists does not contain only `<li>` elements and script supporting elements ' +
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/list.js]: s/does/do

File: lighthouse-core/audits/accessibility/meta-refresh.js [hunk 12]
```diff
@@ -33,6 +33,7 @@ class MetaRefresh extends AxeAudit {
       category: 'Accessibility',
       name: 'meta-refresh',
       description: 'The document does not use `<meta http-equiv="refresh">`.',
+      fallbackDescription: 'The document use `<meta http-equiv="refresh">`.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/meta-refresh.js]: s/use/uses

File: lighthouse-core/audits/accessibility/meta-viewport.js [hunk 13]
```diff
@@ -34,6 +34,8 @@ class MetaViewport extends AxeAudit {
       name: 'meta-viewport',
       description: '`[user-scalable="no"]` is not used in the `<meta name="viewport">` ' +
           'element and the `[maximum-scale]` attribute is not less than 5.',
+      fallbackDescription: '`[user-scalable="no"]` is used in the `<meta name="viewport">` ' +
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/meta-viewport.js]: s/and/or

File: lighthouse-core/audits/accessibility/object-alt.js [hunk 14]
```diff
@@ -33,6 +33,7 @@ class ObjectAlt extends AxeAudit {
       category: 'Accessibility',
       name: 'object-alt',
       description: '`<object>` elements have `[alt]` text.',
+      fallbackDescription: '`<object>` elements does not have `[alt]` text.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/object-alt.js]: s/does/do

File: lighthouse-core/audits/accessibility/td-headers-attr.js [hunk 15]
```diff
@@ -35,6 +35,8 @@ class TDHeadersAttr extends AxeAudit {
       name: 'td-headers-attr',
       description: 'Cells in a `<table>` element that use the `[headers]` attribute only refer ' +
           'to other cells of that same table.',
+      fallbackDescription: 'Cells in a `<table>` element that use the `[headers]` attribute does not only refer ' +
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/td-headers-attr.js]: s/does not only refer/refers

File: lighthouse-core/audits/accessibility/th-has-data-cells.js [hunk 16]
```diff
@@ -34,6 +34,8 @@ class THHasDataCells extends AxeAudit {
       name: 'th-has-data-cells',
       description: '`<th>` elements and elements with `[role="columnheader"/"rowheader"]` have ' +
           'data cells they describe.',
+      fallbackDescription: '`<th>` elements and elements with `[role="columnheader"/"rowheader"]` does not have ' +
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/th-has-data-cells.js]: s/does/do

File: lighthouse-core/audits/accessibility/valid-lang.js [hunk 17]
```diff
@@ -33,6 +33,7 @@ class ValidLang extends AxeAudit {
       category: 'Accessibility',
       name: 'valid-lang',
       description: '`[lang]` attributes have a valid value.',
+      fallbackDescription: '`[lang]` attributes does not have a valid value.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/valid-lang.js]: s/does/do

File: lighthouse-core/audits/accessibility/video-caption.js [hunk 18]
```diff
@@ -33,6 +33,7 @@ class VideoCaption extends AxeAudit {
       category: 'Accessibility',
       name: 'video-caption',
       description: '`<video>` elements contain a `<track>` element with `[kind="captions"]`.',
+      fallbackDescription: '`<video>` elements does not contain a `<track>` element with `[kind="captions"]`.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/video-caption.js]: s/does/do

File: lighthouse-core/audits/accessibility/video-description.js [hunk 19]
```diff
@@ -33,6 +33,7 @@ class VideoDescription extends AxeAudit {
       category: 'Accessibility',
       name: 'video-description',
       description: '`<video>` elements contain a `<track>` element with `[kind="description"]`.',
+      fallbackDescription: '`<video>` elements does not contain a `<track>` element with `[kind="description"]`.',
```
Comment (patrickhulce) [inline lighthouse-core/audits/accessibility/video-description.js]: s/does/do

File: lighthouse-core/audits/audit.js [hunk 20]
```diff
@@ -136,7 +137,18 @@ class Audit {
     if (displayValue === score) {
       displayValue = '';
     }
+    let auditDescription = audit.meta.description;
 
+    if((result.error || result.rawValue === false) && audit.meta.fallbackDescription) {
```
Comment (patrickhulce) [inline lighthouse-core/audits/audit.js]: let's use the same logic we do for display, the logic applies if `!result.rawValue || (typeof result.rawValue === 'number' && result.rawValue < 100)`

File: lighthouse-core/audits/byte-efficiency/total-byte-weight.js [hunk 21]
```diff
@@ -39,6 +39,7 @@ class TotalByteWeight extends ByteEfficiencyAudit {
       name: 'total-byte-weight',
       optimalValue: `< ${this.bytesToKbString(OPTIMAL_VALUE)}`,
       description: 'Avoids enormous network payloads',
+      fallbackDescription: 'Does not avoid enormous network payloads',
```
Comment (patrickhulce) [inline lighthouse-core/audits/byte-efficiency/total-byte-weight.js]: s/Does not avoid/Has

File: lighthouse-core/audits/consistently-interactive.js [hunk 22]
```diff
@@ -41,6 +41,7 @@ class ConsistentlyInteractiveMetric extends Audit {
       category: 'Performance',
       name: 'consistently-interactive',
       description: 'Consistently Interactive (beta)',
+      fallbackDescription: 'Is not Consistently Interactive (beta)',
```
Comment (patrickhulce) [inline lighthouse-core/audits/consistently-interactive.js]: remove

File: lighthouse-core/audits/deprecations.js [hunk 23]
```diff
@@ -34,6 +34,7 @@ class Deprecations extends Audit {
       category: 'Deprecations',
       name: 'deprecations',
       description: 'Avoids deprecated APIs',
+      fallbackDescription: 'Does not avoid deprecated API\'s',
```
Comment (patrickhulce) [inline lighthouse-core/audits/deprecations.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/appcache-manifest.js [hunk 24]
```diff
@@ -33,6 +33,7 @@ class AppCacheManifestAttr extends Audit {
       category: 'Offline',
       name: 'appcache-manifest',
       description: 'Avoids Application Cache',
+      fallbackDescription: 'Does not avoid Application Cache',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/appcache-manifest.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/dom-size.js [hunk 25]
```diff
@@ -48,6 +48,7 @@ class DOMSize extends Audit {
       category: 'Performance',
       name: 'dom-size',
       description: 'Avoids an excessive DOM size',
+      fallbackDescription: 'Does not avoid an excessive DOM size',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/dom-size.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/geolocation-on-start.js [hunk 26]
```diff
@@ -34,6 +34,7 @@ class GeolocationOnStart extends ViolationAudit {
       category: 'UX',
       name: 'geolocation-on-start',
       description: 'Avoids requesting the geolocation permission on page load',
+      fallbackDescription: 'Does not avoid requesting the geolocation permission on page load',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/geolocation-on-start.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/no-document-write.js [hunk 27]
```diff
@@ -34,6 +34,7 @@ class NoDocWriteAudit extends ViolationAudit {
       category: 'Performance',
       name: 'no-document-write',
       description: 'Avoids `document.write()`',
+      fallbackDescription: 'Does not avoid `document.write()`',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/no-document-write.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/no-mutation-events.js [hunk 28]
```diff
@@ -51,6 +51,7 @@ class NoMutationEventsAudit extends Audit {
       category: 'JavaScript',
       name: 'no-mutation-events',
       description: 'Avoids Mutation Events in its own scripts',
+      fallbackDescription: 'Does not avoid Mutation Events in its own scripts',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/no-mutation-events.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/no-old-flexbox.js [hunk 29]
```diff
@@ -37,6 +37,7 @@ class NoOldFlexboxAudit extends Audit {
       category: 'CSS',
       name: 'no-old-flexbox',
       description: 'Avoids old CSS flexbox',
+      fallbackDescription: 'Does not avoid old CSS flexbox',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/no-old-flexbox.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/no-websql.js [hunk 30]
```diff
@@ -34,6 +34,7 @@ class NoWebSQLAudit extends Audit {
       category: 'Offline',
       name: 'no-websql',
       description: 'Avoids WebSQL DB',
+      fallbackDescription: 'Does not avoid WebSQL DB',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/no-websql.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/dobetterweb/notification-on-start.js [hunk 31]
```diff
@@ -34,6 +34,7 @@ class NotificationOnStart extends ViolationAudit {
       category: 'UX',
       name: 'notification-on-start',
       description: 'Avoids requesting the notification permission on page load',
+      fallbackDescription: 'Does not avoid requesting the notification permission on page load',
```
Comment (patrickhulce) [inline lighthouse-core/audits/dobetterweb/notification-on-start.js]: s/Does not avoid/Uses

File: lighthouse-core/audits/accessibility/aria-allowed-attr.js [hunk 32]
```diff
@@ -33,6 +33,7 @@ class ARIAAllowedAttr extends AxeAudit {
       category: 'Accessibility',
       name: 'aria-allowed-attr',
       description: '`[aria-*]` attributes match their roles.',
+      fallbackDescription: '`[aria-*]` attributes does not match their roles.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/aria-allowed-attr.js]: do not match

File: lighthouse-core/audits/accessibility/aria-required-attr.js [hunk 33]
```diff
@@ -33,6 +33,7 @@ class ARIARequiredAttr extends AxeAudit {
       category: 'Accessibility',
       name: 'aria-required-attr',
       description: '`[role]`s have all required `[aria-*]` attributes.',
+      fallbackDescription: '`[role]`s does not have all required `[aria-*]` attributes.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/aria-required-attr.js]: do not have all required...

File: lighthouse-core/audits/accessibility/aria-required-children.js [hunk 34]
```diff
@@ -34,6 +34,7 @@ class AriaRequiredChildren extends AxeAudit {
       category: 'Accessibility',
       name: 'aria-required-children',
       description: '`[role]`s that require child `[role]`s contain them.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/aria-required-children.js]: let's change these to:

```
description: 'Elements with `[role]` that require specific children `[role]`s, are present.',
fallbackDescription: 'Elements with `[role]` that require specific children `[role]`s, are missing.',
```

File: lighthouse-core/audits/accessibility/aria-valid-attr-value.js [hunk 35]
```diff
@@ -33,6 +33,7 @@ class ARIAValidAttr extends AxeAudit {
       category: 'Accessibility',
       name: 'aria-valid-attr-value',
       description: '`[aria-*]` attributes have valid values.',
+      fallbackDescription: '`[aria-*]` attributes does not have valid values.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/aria-valid-attr-value.js]: do not have

File: lighthouse-core/audits/accessibility/aria-valid-attr.js [hunk 36]
```diff
@@ -33,6 +33,7 @@ class ARIAValidAttr extends AxeAudit {
       category: 'Accessibility',
       name: 'aria-valid-attr',
       description: '`[aria-*]` attributes are valid and not misspelled.',
+      fallbackDescription: '`[aria-*]` attributes are not valid and misspelled.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/aria-valid-attr.js]: are not valid or misspelled

File: lighthouse-core/audits/accessibility/audio-caption.js [hunk 37]
```diff
@@ -33,6 +33,7 @@ class AudioCaption extends AxeAudit {
       category: 'Accessibility',
       name: 'audio-caption',
       description: '`<audio>` elements contain a `<track>` element with `[kind="captions"]`.',
+      fallbackDescription: '`<audio>` elements does not contain a `<track>` element with `[kind="captions"]`.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/audio-caption.js]: elements does not contain -> elements are missing a

File: lighthouse-core/audits/accessibility/button-name.js [hunk 38]
```diff
@@ -33,6 +33,7 @@ class ButtonName extends AxeAudit {
       category: 'Accessibility',
       name: 'button-name',
       description: 'Buttons have an accessible name.',
+      fallbackDescription: 'Buttons does not have an accessible name.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/button-name.js]: do not have

File: lighthouse-core/audits/accessibility/definition-list.js [hunk 39]
```diff
@@ -34,6 +34,8 @@ class DefinitionList extends AxeAudit {
       name: 'definition-list',
       description: '`<dl>`\'s contain only properly-ordered `<dt>` and `<dd>` groups, `<script>` ' +
           'or <template> elements.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/definition-list.js]: `<template>` -> \``<template>`\` for both

File: lighthouse-core/audits/accessibility/tabindex.js [hunk 40]
```diff
@@ -33,6 +33,7 @@ class TabIndex extends AxeAudit {
       category: 'Accessibility',
       name: 'tabindex',
       description: 'No element has a `[tabindex]` value greater than 0.',
+      fallbackDescription: 'One or more elements has a `[tabindex]` value greater than 0.',
```
Comment (ebidel) [inline lighthouse-core/audits/accessibility/tabindex.js]: One or more -> Some

has -> have

File: lighthouse-core/audits/audit.js [hunk 41]
```diff
@@ -136,7 +137,18 @@ class Audit {
...
+      auditDescription = audit.meta.fallbackDescription;
+    }
+    if(typeof result.rawValue !== 'boolean' && audit.meta.scoringMode === 'numeric' && audit.meta.fallbackDescription) {
```
Comment (ebidel) [inline lighthouse-core/audits/audit.js]: space:` if (`

File: lighthouse-core/audits/audit.js [hunk 42]
```diff
@@ -136,7 +137,18 @@ class Audit {
...
+      let fallbackAuditDescriptionScore = Util.calculateRating(score)
+      // TODO: Here we could also add a average description if needed
+      if(fallbackAuditDescriptionScore === 'fail') {
```
Comment (ebidel) [inline lighthouse-core/audits/audit.js]: `if (`

File: lighthouse-core/audits/audit.js (hunk 20 above)
Comment (ebidel) [inline lighthouse-core/audits/audit.js]: With this, we only need one of these checks to set the `auditDescription`. Right? Why are there two here?

File: lighthouse-core/audits/dobetterweb/uses-http2.js [hunk 43]
```diff
@@ -36,6 +36,7 @@ class UsesHTTP2Audit extends Audit {
       category: 'Performance',
       name: 'uses-http2',
       description: 'Uses HTTP/2 for its own resources',
+      fallbackDescription: 'Does not use HTTP/2 for its own resources',
```
Comment (ebidel) [inline lighthouse-core/audits/dobetterweb/uses-http2.js]: Does not use HTTP/2 for all of its resources

File: lighthouse-core/audits/is-on-https.js [hunk 44]
```diff
@@ -32,6 +32,7 @@ class HTTPS extends Audit {
       category: 'Security',
       name: 'is-on-https',
       description: 'Uses HTTPS',
+      fallbackDescription: 'Does not use HTTPS',
```
Comment (ebidel) [inline lighthouse-core/audits/is-on-https.js]: remove

File: lighthouse-core/audits/themed-omnibox.js [hunk 45]
```diff
@@ -29,6 +29,7 @@ class ThemedOmnibox extends MultiCheckAudit {
       category: 'PWA',
       name: 'themed-omnibox',
       description: 'Address bar matches brand colors',
+      fallbackDescription: 'Address bar does not match brand color',
```
Comment (ebidel) [inline lighthouse-core/audits/themed-omnibox.js]: colors

File: lighthouse-core/audits/webapp-install-banner.js [hunk 46]
```diff
@@ -39,6 +39,7 @@ class WebappInstallBanner extends MultiCheckAudit {
       category: 'PWA',
       name: 'webapp-install-banner',
       description: 'User can be prompted to Install the Web App',
+      fallbackDescription: 'User cannot be prompted to Install the Web App',
```
Comment (ebidel) [inline lighthouse-core/audits/webapp-install-banner.js]: User won't be prompted....

File: lighthouse-core/audits/without-javascript.js [hunk 47]
```diff
@@ -28,6 +28,7 @@ class WithoutJavaScript extends Audit {
       category: 'JavaScript',
       name: 'without-javascript',
       description: 'Contains some content when JavaScript is not available',
+      fallbackDescription: 'Does not contain some content when JavaScript is not available',
```
Comment (ebidel) [inline lighthouse-core/audits/without-javascript.js]: Does not provide fallback content when JavaScript is not available

**Comment by ev1stensberg** [comment]
@ebidel @patrickhulce Will change this tomorrow, hard to keep track of everything at once! Sorry!

File: lighthouse-core/audits/audit.js (hunk 20 above)
Comment (ev1stensberg) [inline lighthouse-core/audits/audit.js]: Yeah, sorry, was to split up the logic abit to get feedback + simplify for initial iteration

File: lighthouse-core/audits/accessibility/meta-viewport.js (hunk 13 above)
Comment (ev1stensberg) [inline lighthouse-core/audits/accessibility/meta-viewport.js]: This should be changed now, mind having a second look?

**Comment by ev1stensberg** [comment]
Let's discuss some logic in `audits` for the determination of displaying `fallbackDescription`. Was this implementation something similar as you had in mind? Also, need feedback on how to test CSS coverage, didn't even know this was implemented yet ( super awesome! )!

**Comment by ev1stensberg** [comment]
Also, sorry for the grammar mistakes, pulled this together very late yesterday. Same for the https redirect PR, prefer iteration before actual implementation :)

**Comment by ev1stensberg** [comment]
We can also change the help text to give some perf advice btw, could be a good way to get beginners understanding what they've done wrong more extensively

**Comment by ev1stensberg** [comment]
I'll finish this tomorrow. What is missing besides CSS audits before this is a wrap?

**Pull request closed without merging**